apps/web/src/components/Basenames/UsernamePill/index.tsx

```diff
@@ -37,21 +27,41 @@ export function UsernamePill({ variant, username, address }: UsernamePillProps)
 
   const avatarClasses = classNames(
     'flex items-center justify-center overflow-hidden rounded-full',
-    'absolute',
     transitionClasses,
     {
-      'h-[4rem] max-h-[4rem] min-h-[4rem] w-[4rem] min-w-[4rem] max-w-[4rem] top-4 left-4':
+      'h-[4rem] max-h-[4rem] min-h-[4rem] w-[4rem] min-w-[4rem] max-w-[4rem]':
         variant === UsernamePillVariants.Inline,
       'h-[3rem] max-h-[3rem] min-h-[3rem] w-[3rem] min-w-[3rem] max-w-[3rem] top-10 left-10':
         variant === UsernamePillVariants.Card,
     },
   );
 
+  const userNameFontScale = useMemo(() => {
+    if (username.length >= 25) {
+      return 'text-md';
+    }
+    if (username.length >= 22) {
+      return 'text-lg';
+    }
+    if (username.length >= 19) {
+      return 'text-xl';
+    }
+    if (username.length >= 15) {
+      return 'text-2xl';
+    }
+    // 3 letter name with .base.eth
+    if (username.length >= 12) {
+      return 'text-3xl';
+    }
+    return 'text-3xl';
+  }, [username]);
```

Comment:
How could we combine this with something like `clamp`
https://developer.mozilla.org/en-US/docs/Web/CSS/clamp

to accomplish a length and viewport-dynamic sizing?

Reply:
Nice prompt